### DISCUSSION TITLE
Populate

### DISCUSSION THREAD

**Discussion by Tandome** [google/or-tools#4306] [CP-SAT questions]
Hello all, do you know CpSolver has an equivalent to the 'populate()' method of ILOG CPLEX, which searches all solutions with a gap of tolerance for quasi-optimality ? I can't manage to do something analog (I am solving an ILP problem with CP-SAT).

Thank you very much !

Best,

Edmond

**Comment by lperron** [comment]
You can enumerate all solutions. This does not work with an objective.
If you know the optimal,you can add a constraint on the objective variable to be close to the optimal value, then enumerate all solutions.

**Reply by Tandome** [reply]
Bonjour Laurent,

The optimal value of the objective can change unfortunately... I tried enumerateallsolutions = true, so in that case is it supposed to give all feasible solutions regardless of the objective value ? It seems to stop the search when it finds one optimal solution.

Thanks again !

**Comment by lperron** [comment]
make sure that when enumerate_all_solutions=true, there are no objectives
in the model.

Then it will enumerate all solutions.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00
